mousestyles/classification/classification.py

```diff
@@ -0,0 +1,171 @@
+    return(predict_true)
+
+
+def GetSummary(result):
+    """
+    Returns a ndarray of classification result summary,
+    including precision, recall, F1 measure in terms of different
+    strains.
+    Parameters
+    ----------
+    result: classification results in test data set
+        Column 0: prediction strain labels
+        Column 1: true strain labels
+    Returns
+    ----------
+    ndarray of classification result summary, shape(16,3).
+       16 rows, for each strain 0-15
+       Column 0: precision
+       Column 1: recall
+       Column 2: F-1 measure
+
+    """
+    prediction_accurate_count_matrix = pd.crosstab(index=result.iloc[:, 0],
+                                                   columns=result.iloc[:, 1],
+                                                   margins=True)
+    prediction_accurate_count_matrix.rename(columns={"All": "rowTotal"},
+                                            inplace=True)
+    prediction_accurate_count_matrix.rename(index={"All": "colTotal"},
+                                            inplace=True)
+    prediction_accurate_rate_matrix = prediction_accurate_count_matrix.T /\
+        prediction_accurate_count_matrix["rowTotal"]
+    # get the precision list
+    precision = [prediction_accurate_rate_matrix.T.iloc[i, i]
+                 for i in range(prediction_accurate_rate_matrix.shape[1]-1)]
+    prediction_accurate_rate_matrix = prediction_accurate_count_matrix /\
+        prediction_accurate_count_matrix.ix["colTotal"]
+    # get the recall list
+    recall = [prediction_accurate_rate_matrix.T.iloc[i, i]
+              for i in range(prediction_accurate_rate_matrix.shape[1]-1)]
+    # get the F1 list
+    f1 = [2*precision[i]*recall[i]/(precision[i]+recall[i])
+          for i in range(prediction_accurate_rate_matrix.shape[1]-1)]
+    summary = pd.concat([pd.DataFrame(precision),
+                         pd.DataFrame(recall), pd.DataFrame(f1)], axis=1)
+    summary.columns = ['precision', 'recall', "F1_score"]
+    return(summary)
+
+
+def GradientBoosting(strain, features, rseed=222):
+    """
+    Returns a ndarray of GradientBoosting results, containing prediction strain
+    labels and true strain labels for test data set. Use cross validation
+    to tune Parameters.
+    Parameters
+    ----------
+    strain: ndarray, shape of (1921,)
+    classification labels
+    features: ndarray, shape of (1921, 99)
+    classification features
+    rseed: int, optional
+    random seed for shuffling the data set to separate train and test
+    Returns
+    ----------
+    ndarray of RandomForest results in test data set.
+    Column 0: prediction strain labels
+    Column 1: true strain labels
+    """
+    train_y, train_x, test_y, test_x = prep_data(strain, features, rseed)
+    # creat GradientBoosting model
+    scaler = StandardScaler()
+    train_x = scaler.fit_transform(train_x)
+    es = [200]
+    ls = np.linspace(0.0001, 0.15, 10)
```

Comment:
Same here, we should let user define these parameters and take them as inputs of the function.